query/stdlib/influxdata/influxdb/rules.go

```diff
@@ -873,6 +878,107 @@ func (p PushDownBareAggregateRule) Rewrite(ctx context.Context, pn plan.Node) (p
 	}), true, nil
 }
 
+// PushDownWindowGroupAggregateRule will match the given pattern.
+// ReadGroupPhys |> window |> { min, max, count, sum }
+//
+// This pattern will use the PushDownWindowAggregateRule to determine
+// if the ReadWindowAggregatePhys operation is available before it will
+// rewrite the above. This rewrites the above to:
+//
+// ReadWindowAggregatePhys |> group(columns: ["_start", "_stop", ...]) |> { min, max, sum }
+//
+// The count aggregate uses sum to merge the results.
+type PushDownWindowGroupAggregateRule struct{}
```

Comment:
It just occurs to me that this is just changing from pushing down `group` to pushing down `window`.
I think more precisely I would like to call it `TransposeGroupWindowAggregateRule`, because it transposes ReadGroup-window to be ReadWindow-group.